### PR TITLE
fix: Get-AzCachedAccessToken -AssignGlobalVariables

### LINKED ISSUE #238: Get-AzCachedAccessToken -AssignGlobalVariables does not assign global variables
**Describe the bug**
Get-AzCachedAccessToken -AssignGlobalVariables does not assign global variables
Nor does Get-AzCachedAccessToken -AssignGlobalVariables $True / $False or another variation

**To Reproduce**
Connect to cloud shell and execute Get-AzCachedAccessToken -AssignGlobalVariables
In the output you can also see that the output produced from following lines never show up

    $Global:subscriptionId = $currentAzureContext.Subscription.Id
    Write-Host "Global variable 'subscriptionId' assigned"

    $Global:accessToken = $token.AccessToken
    Write-Host "Global variable 'accessToken' assigned"

**Expected behavior**
Global variables to be assigned when you use Get-AzCachedAccessToken -AssignGlobalVariables, or global variables to be assigned when you Get-AzCachedAccessToken -AssignGlobalVariables $True (since documentation states that it is a switch default false)

**Additional context**
 - Using latest version

### PR DESCRIPTION
**Pull request by spasal**
Updated script and documentation to fix Get-AzCachedAccessToken -AssignGlobalVariables

Closes #238

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Thx for the PR! 
We should make sure that we test this thoroughly. Could you update both the unit and integration tests? Thx!

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
- [x] Add integration test to verify that with a valid Azure connection we can assign variables globally
- [x] Add/adapt unit test to make sure we test the global assignment and switch parameter

**Review by fgheysels — APPROVED** [review]
Thanks for your contribution!

**Comment by fgheysels** [comment]
> Thx for the PR! We should make sure that we test this thoroughly. Could you update both the unit and integration tests? Thx!

I might be wrong, but the tests were already there.  [They're already using this parameter for the changes that Alex has made](https://github.com/arcus-azure/arcus.scripting/blob/62568cb0e5589421e47e86b9cdd2fe98825f95f3/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L113).   (At least, afaik @spasal ? )

**Comment by stijnmoreels** [comment]
> > Thx for the PR! We should make sure that we test this thoroughly. Could you update both the unit and integration tests? Thx!
> 
> I might be wrong, but the tests were already there. [They're already using this parameter for the changes that Alex has made](https://github.com/arcus-azure/arcus.scripting/blob/62568cb0e5589421e47e86b9cdd2fe98825f95f3/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L113). (At least, afaik @spasal ? )
> 
> otoh, when I read the related issue, it seems like this behavior is indeed not correctly tested by the tests we currently have.

The tests are indeed not correct, and the integration test for this is missing. In any case, this bug-fix PR should have a change in tests to 'prove' it is fixed now 😉 .

**Comment by spasal** [comment]
> > Thx for the PR! We should make sure that we test this thoroughly. Could you update both the unit and integration tests? Thx!
> 
> I might be wrong, but the tests were already there. [They're already using this parameter for the changes that Alex has made](https://github.com/arcus-azure/arcus.scripting/blob/62568cb0e5589421e47e86b9cdd2fe98825f95f3/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L113). (At least, afaik @spasal ? )
> 
> otoh, when I read the related issue, it seems like this behavior is indeed not correctly tested by the tests we currently have.

Indeed, quickly took a look yesterday and it should be already tested in the unit tests and not yet in the integration tests. 

As the integration tests were passing (even though it was not working), those need to be updated. Integration tests need to be added. As I am new to Pester it may take a bit to get the hang of it, will try it out!

**Comment by spasal** [comment]
> * [x]  Add integration test to verify that with a valid Azure connection we can assign variables globally
> * [x]  Add/adapt unit test to make sure we test the global assignment and switch parameter

@stijnmoreels unit tests were already present and are working as expected. Integration tests for this case have been added

File: docs/preview/02-Features/powershell/azure-security.md [hunk 1]
```diff
@@ -58,8 +58,7 @@ When you want to make use of the REST-API's made available to manage Azure Resou
 
 | Parameter               | Mandatory | Description                                                                                       |
 | ----------------------- | --------- | ------------------------------------------------------------------------------------------------- |
-| `AssignGlobalVariables` | no        | Switch - default value: false                                                                     |
-|                         |           | When you want the global variables `access_token` and `subscriptionId` assigned for easy access.  |
+| `AssignGlobalVariables` | no        | Indicator (switch) whether you want the global variables `access_token` and `subscriptionId` assigned for easy access.  |
```
Comment (stijnmoreels) [inline docs/preview/02-Features/powershell/azure-security.md:61]: We should mention that the default is `false`.

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Security.tests.ps1 [hunk 2]
```diff
@@ -16,9 +16,23 @@ InModuleScope Arcus.Scripting.Security {
                 $token.SubscriptionId | Should -Be $config.Arcus.SubscriptionId
                 $token.AccessToken | Should -Not -BeNullOrEmpty
             }
+            It "Get cached access token from current active authenticated Azure session succeeds with assigning global variables" {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Security.tests.ps1:19]: Yes, great, this is good 👍 .

**Comment by stijnmoreels** [comment]
> > * [x]   Add integration test to verify that with a valid Azure connection we can assign variables globally
> > * [x]   Add/adapt unit test to make sure we test the global assignment and switch parameter
> 
> @stijnmoreels unit tests were already present and are working as expected. Integration tests for this case have been added

THe unit tests are not really correctly implemented as the stubbed function already assigned the gobal veriables. We should update that test too.
https://github.com/arcus-azure/arcus.scripting/blob/d996cde664f893ca8a45ce4b1161e2aa9eaed589/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L106

File: docs/preview/02-Features/powershell/azure-security.md (hunk 1 above)
Comment (spasal) [reply docs/preview/02-Features/powershell/azure-security.md:61]: Something like this?
Indicator (switch - default value: false) whether you want the global variables `access_token` and `subscriptionId` assigned for easy access.

**Comment by spasal** [comment]
> > > * [x]    Add integration test to verify that with a valid Azure connection we can assign variables globally
> > > * [x]    Add/adapt unit test to make sure we test the global assignment and switch parameter
> > 
> > 
> > @stijnmoreels unit tests were already present and are working as expected. Integration tests for this case have been added
> 
> THe unit tests are not really correctly implemented as the stubbed function already assigned the gobal veriables. We should update that test too.
> 
> https://github.com/arcus-azure/arcus.scripting/blob/d996cde664f893ca8a45ce4b1161e2aa9eaed589/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L106

Not really sure what we can do except that the stubbed function assigns the global variables. Any ideas on how to improve?

File: docs/preview/02-Features/powershell/azure-security.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/02-Features/powershell/azure-security.md:61]: Yes, that's ok. 👍

**Comment by stijnmoreels** [comment]
> > > > * [x]     Add integration test to verify that with a valid Azure connection we can assign variables globally
> > > > * [x]     Add/adapt unit test to make sure we test the global assignment and switch parameter
> > > 
> > > 
> > > @stijnmoreels unit tests were already present and are working as expected. Integration tests for this case have been added
> > 
> > 
> > THe unit tests are not really correctly implemented as the stubbed function already assigned the gobal veriables. We should update that test too.
> > https://github.com/arcus-azure/arcus.scripting/blob/d996cde664f893ca8a45ce4b1161e2aa9eaed589/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1#L106
> 
> Not really sure what we can do except that the stubbed function assigns the global variables. Any ideas on how to improve?

Good job finding the real issue here 👍

File: src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1 [hunk 3]
```diff
@@ -29,9 +29,20 @@ Export-ModuleMember -Function Remove-AzResourceGroupLocks
...
+ .Parameter AssignGlobalVariables
+  Indicate whether to assign the global variables `access_token` and `subscriptionId` for easy access.
 #>
 function Get-AzCachedAccessToken {
-    . $PSScriptRoot\Scripts\Get-AzCachedAccessToken.ps1
+    param(
+        [Parameter(Mandatory = $false)][switch] $AssignGlobalVariables = $false
+    )
+
+    if($AssignGlobalVariables) {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1:41]: Nit - code style we use
```suggestion
    if ($AssignGlobalVariables) {
```

**Review by stijnmoreels — APPROVED** [review]
Ok, great. I think we can almost complete this. Nice job! 💯

**Comment by stijnmoreels** [comment]
Thanks again for noticing and fixing this 😉👌

**Pull request merged**